### PR TITLE
test: List v2 meta widget generator jest test

### LINKED ISSUE #16788: Task: List v2 - Jest tests for MetaWidgetGenerator

### PR DESCRIPTION
**Pull request by ashit-rath**
## Description

Jest test for meta widget generator and minor refactor.

Fixes #16788

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change
- Jest tests

## How Has This Been Tested?

- Manual
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.test.ts [hunk 1]
```diff
@@ -0,0 +1,891 @@
...
+        occurrence: 2,
+      },
+      {
+        widgetType: "INPUT_WIDGET_V2",
+        occurrence: 1,
+      },
+    ]);
+  });
+
+  it("it does not re-generates meta widgets options don't change", () => {
```
Comment (Tooluloope) [inline app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.test.ts:291]:
```suggestion
  it("it does not re-generates meta widgets when options don't change", () => {
```

File: app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.test.ts [hunk 2]
```diff
@@ -0,0 +1,891 @@
...
+        occurrence: 2,
+      },
+      {
+        widgetType: "CONTAINER_WIDGET",
+        occurrence: 2,
+      },
+    ]);
+  });
+
+  it("it re-generates meta widgets when template widgets page no changes", () => {
```
Comment (Tooluloope) [inline app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.test.ts:378]:
```suggestion
  it("it re-generates meta widgets when List widget page no changes", () => {
```

File: app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.test.ts [hunk 3]
```diff
@@ -0,0 +1,891 @@
...
+        occurrence: 4,
+      },
+      {
+        widgetType: "INPUT_WIDGET_V2",
+        occurrence: 2,
+      },
+    ]);
+  });
+
+  it("it generates only extra meta widgets when template widgets page size increases", () => {
```
Comment (Tooluloope) [inline app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.test.ts:413]:
```suggestion
  it("it generates only extra meta widgets when List widget's page size increases", () => {
```

File: app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.test.ts [hunk 4]
```diff
@@ -0,0 +1,891 @@
...
+        occurrence: 2,
+      },
+      {
+        widgetType: "INPUT_WIDGET_V2",
+        occurrence: 1,
+      },
+    ]);
+  });
+
+  it("it removes meta widgets when template widgets page size decreases", () => {
```
Comment (Tooluloope) [inline app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.test.ts:448]:
```suggestion
  it("it removes meta widgets when List widget's page size decreases", () => {
```

**Review by Tooluloope — CHANGES_REQUESTED** [review]
The Properties belong to List widget not template widget

File: app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.test.ts (hunk 2 above)
Comment (ashit-rath) [reply app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.test.ts:378]: @Tooluloope I think we should just mention
`it re-generates meta widgets when page no changes`

Comment (Tooluloope) [reply app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.test.ts:378]: Yes that's better @ashit-rath

**Review by Tooluloope — APPROVED** [review]

**Pull request merged**